gandi/cli/commands/ip.py

```diff
@@ -199,52 +217,65 @@ def create(gandi, datacenter, bandwidth, ip_version, vlan, ip, attach,
 
 
 @cli.command()
-@click.argument('resource')
+@click.argument('resources', nargs=-1)
 @click.option('--bg', '--background', default=False, is_flag=True,
               help='Run command in background mode (default=False).')
 @click.option('--force', '-f', is_flag=True,
               help='This is a dangerous option that will cause CLI to continue'
                    ' without prompting. (default=False).')
 @pass_gandi
-def detach(gandi, resource, background, force):
-    """Detach an ip from it's currently attached vm.
+def detach(gandi, resources, background, force):
+    """Detach one or more IPs from the vm they are attached to.
 
     resource can be an ip id or ip.
     """
-    if not force:
-        proceed = click.confirm('Are you sure to detach ip %s?' % resource)
-        if not proceed:
-            return
+    resources = sorted(tuple(set(resources)))
+    ret = []
 
-    return gandi.ip.detach(resource, background, force)
+    for num, item in enumerate(resources):
+        if not force:
```

Comment:
same remarks as in attach, you should check for force outside the loop, and group the operations into one `IP.detach` call.